core/lz4/src/ZipLZ4.cxx

```diff
@@ -0,0 +1,139 @@
+// Original Author: Brian Bockelman
+
+/*************************************************************************
+ * Copyright (C) 1995-2017, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "ZipLZ4.h"
+#include "lz4.h"
+#include "lz4hc.h"
+#include <stdio.h>
+#include <cinttypes>
+#include <cstdint>
+#include <cstring>
+
+#include "RConfig.h"
+
+// Pulled from liblz4; upstream library explicitly exposes the symbol but the default build
+// excludes the header.
+typedef unsigned long long XXH64_hash_t;
+typedef struct {
+   unsigned char digest[8];
+} XXH64_canonical_t;
+extern "C" XXH64_hash_t LZ4_XXH64(const void *input, size_t length, unsigned long long seed);
```

Comment:
Those seems to be copy/paste from lz4 headers .. they should be explicitly commented as such (this actually might invalidate also the first couple of comments).

An alternative is to provide 'small' wrapper functions so that the real header can be used (in the source file).  However this might have an unacceptable performance penalty ...